ulab_Crunch_Numbers_Fast/waterfall/code.py

```diff
@@ -12,32 +12,81 @@
 import audiobusio
 import displayio
 from ulab import numpy as np
-from ulab.scipy.signal import spectrogram
+
+try:
+    from ulab.utils import spectrogram
+except ImportError:
+    from ulab.scipy.signal import spectrogram
 
 display = board.DISPLAY
 
 # Create a heatmap color palette
 palette = displayio.Palette(52)
-for i, pi in enumerate((0xff0000, 0xff0a00, 0xff1400, 0xff1e00,
-                        0xff2800, 0xff3200, 0xff3c00, 0xff4600,
-                        0xff5000, 0xff5a00, 0xff6400, 0xff6e00,
-                        0xff7800, 0xff8200, 0xff8c00, 0xff9600,
-                        0xffa000, 0xffaa00, 0xffb400, 0xffbe00,
-                        0xffc800, 0xffd200, 0xffdc00, 0xffe600,
-                        0xfff000, 0xfffa00, 0xfdff00, 0xd7ff00,
-                        0xb0ff00, 0x8aff00, 0x65ff00, 0x3eff00,
-                        0x17ff00, 0x00ff10, 0x00ff36, 0x00ff5c,
-                        0x00ff83, 0x00ffa8, 0x00ffd0, 0x00fff4,
-                        0x00a4ff, 0x0094ff, 0x0084ff, 0x0074ff,
-                        0x0064ff, 0x0054ff, 0x0044ff, 0x0032ff,
-                        0x0022ff, 0x0012ff, 0x0002ff, 0x0000ff)):
-    palette[51-i] = pi
+for i, pi in enumerate(
+    (
+        0xFF0000,
+        0xFF0A00,
+        0xFF1400,
+        0xFF1E00,
+        0xFF2800,
+        0xFF3200,
+        0xFF3C00,
+        0xFF4600,
+        0xFF5000,
+        0xFF5A00,
+        0xFF6400,
+        0xFF6E00,
+        0xFF7800,
+        0xFF8200,
+        0xFF8C00,
+        0xFF9600,
+        0xFFA000,
+        0xFFAA00,
+        0xFFB400,
+        0xFFBE00,
+        0xFFC800,
+        0xFFD200,
+        0xFFDC00,
+        0xFFE600,
+        0xFFF000,
+        0xFFFA00,
+        0xFDFF00,
+        0xD7FF00,
+        0xB0FF00,
+        0x8AFF00,
+        0x65FF00,
+        0x3EFF00,
+        0x17FF00,
+        0x00FF10,
+        0x00FF36,
+        0x00FF5C,
+        0x00FF83,
+        0x00FFA8,
+        0x00FFD0,
+        0x00FFF4,
+        0x00A4FF,
+        0x0094FF,
+        0x0084FF,
+        0x0074FF,
+        0x0064FF,
+        0x0054FF,
+        0x0044FF,
+        0x0032FF,
+        0x0022FF,
+        0x0012FF,
+        0x0002FF,
+        0x0000FF,
+    )
+):
```

Comment:
Please revert this change. Then disable Black for this section, and reenable it after the end of the section.